.pre-commit-config.yaml

```diff
@@ -20,11 +20,11 @@ repos:
     - id: typos
       exclude: _.*_dict.py
 - repo: https://github.com/astral-sh/ruff-pre-commit
-  # Ruff version.
-  rev: v0.5.1
+  rev: v0.9.1
   hooks:
   - id: ruff
-    args: [src, --fix, --show-fixes, --exit-non-zero-on-fix]
+    files: ^src/
```

Comment:
This would not work because pre-commit explicitly passes all files to ruff. Do not tell ruff which files to lint, instead tell pre-commit which files to pass to ruff.

Reply:
I think we already discussed this, but I forgot: why we have 2 separate configs for ruff, both here and in `pyproject.toml`?

Reply:
Indeed. See  #2202.